sensor-gateway/sensor-communication/SensorCommunicator.hpp

```diff
@@ -49,34 +56,50 @@ namespace SensorAccessLinkElement {
 
         SensorCommunicator& operator=(SensorCommunicator&& other)& noexcept = delete;
 
-        void connect() {
+        void start() {
```

Comment:
why change to start?

Reply:
connect etais plus explicite que start apres avec terminateAndJoin ca fait un peu trop thread. Ou il y a une volonté de le faire avec des variables qui porte ces noms?

Reply:
@RphlBrdt `terminateAndJoin` qui fait un peu trop thread c'est specifiquement le but. Cette classe roule principalement un thread.  
@Sebdevar `start` parce qu'apres avoir fait l'appel a `connect` (old) ou  `start` (new), le `SensorCommunicator` :  
 - Se connecte au capteur  
 - Commence a envoyer des donnees dans son ptit thread.  
  
-----  
#### Petite disgression...  
Bref presentement la fonction fait deux trucs, mais `start` explicite mieux le comportement threade de la classe.  
C'est effectivement dans ce cas si une fonction qui fait deux trucs, ce qui n'est pas particulierement clean, good catch you two!  :clinking_glasses:   

Dans le futur ces deux trucs la vont fort certaintement devenir separe quand il va y avoir une affaire comme un `SensorAccessLinkManager` (or something like that)...  

Ca va permettre de permettre une meilleure gestion de reconnection automatique, gestion des erreurs de connection, etc...   
 
----  
Mais pour l'instant ca reste comme ca et la raison principale est celle que @RphlBrdt a point out, ca fait plus thread et c'est exactement le feeling que je veux que ca donne.  
Presentement dans les libs qui l'utilise (surtout pour ROS en fait) c'est plus obvious de `start` un `<sensor>AccessLink` et d'ensuite recevoir des donnees que de simplement le connecter (ce qui garanti pas vraiment un flux de donnees).